src/jit/liveness.cpp

```diff
@@ -1103,6 +1103,13 @@ void Compiler::fgExtendDbgLifetimes()
             {
                 continue;
             }
+#if !defined(_TARGET_64BIT_) && !defined(LEGACY_BACKEND)
+            // On x86, this applies to long-typed variables as well once we've hit LIR. 
```

Comment:
I think this is the right solution for the near term, but I believe that we need to enable long fields of structs to be enregistered, which means addressing the problem of expanding the list of promoted fields for a struct - or coming up with some other clever solution ;-)